### PR TITLE
Core: Fix Generate's slot parsing to default unknown slot names to file name

### PR DESCRIPTION
**Pull request by qwint**
## What is this fixing or adding?
make Generate handle slots without names defined better; fix existing code that seemed to want to handle missing names but was no longer functioning correctly

if there is a better solution than what the existing code was trying to do then i differ to that

## How was this tested?
https://discord.com/channels/731205301247803413/1273345114198118502
tested with the original provided yaml and with a copy that moved the slot without a name to the bottom of the yaml and confirmed with pdb that the args.name got created properly and defaulted to file name when name was missing
also tested with two slots missing name in the same yaml and saw it throw an error mentioning the duplicate names, not sure if that's the best error handling but it works enough for me

### CODE REVIEW

File: Generate.py [hunk 1]
```diff
@@ -200,9 +200,11 @@ def main(args=None) -> Tuple[argparse.Namespace, int]:
...
                                 raise Exception(f"Error setting {k} to {v} for player {player}") from e
 
+                    if not getattr(erargs, "name", None):
+                        setattr(erargs, "name", {})  # if the first slot does not define a name we won't have a name obj to look up
                     if path == args.weights_file_path:  # if name came from the weights file, just use base player name
                         erargs.name[player] = f"Player{player}"
-                    elif not erargs.name[player]:  # if name was not specified, generate it from filename
+                    elif player not in erargs.name:  # if name was not specified, generate it from filename
                         erargs.name[player] = os.path.splitext(os.path.split(path)[-1])[0]
                     erargs.name[player] = handle_name(erargs.name[player], player, name_counter)
```
Comment (Exempt-Medic) [inline Generate.py:209]: Unsure if you'd want to add this to this PR, but this does technically change the fact that previously you could name yourself `false` (which would get changed to the yaml name) and now you cannot name yourself anything boolean in yaml. Could add this to fix it, but maybe there's a better way?
```suggestion
                    erargs.name[player] = handle_name(str(erargs.name[player]), player, name_counter)
```

**Review by Exempt-Medic — APPROVED** [review]
The changes LGTM. Tested them with various configurations of yamls, weights, names/namelessness, and errors were improved and the renaming now works. One small comment on something that's probably out of scope.

File: Generate.py (hunk 1 above)
Comment (qwint) [reply Generate.py:209]: what is expected / what currently works? if you name yourself "false" it currently ignores the name and runs through this code to get the filename instead?

Comment (Exempt-Medic) [reply Generate.py:209]: If you name yourself `false` (no quotation marks) currently it ends up actually hitting the yaml name rename line because your name evaluates as, well, False. This is clearly unintended, but the name does work just fine on WebHost, so I guess it converts to a string somewhere. I think it would make sense to do the same here so that `name: false`, `name: on` etc. will gen instead of crashing

File: Generate.py [hunk 2]
```diff
@@ -200,9 +200,11 @@ def main(args=None) -> Tuple[argparse.Namespace, int]:
                             except Exception as e:
                                 raise Exception(f"Error setting {k} to {v} for player {player}") from e
 
+                    if not getattr(erargs, "name", None):
+                        setattr(erargs, "name", {})  # if the first slot does not define a name we won't have a name obj to look up
```
Comment (NewSoupVi) [inline Generate.py:204]: Somewhat of a nitpick, but I feel like there should just be a `setattr(erargs, "name", {})` before the while loop

**Review by NewSoupVi — APPROVED** [review]
Seems reasonable, I had a comment

File: Generate.py (hunk 2 above)
Comment (qwint) [reply Generate.py:204]: iirc if any yaml has a real name the dict will be made already, so doing that will override existing names
but if none do the attr doesn't exist so the lookups fail, hence the `if`
if you want me to just move the two lines i have no issues though

Comment (qwint) [reply Generate.py:204]: confirmed that this is the case, so unless you just want the names dict blindly initialized in the erargs namespace earlier this is not a simple fix

Comment (NewSoupVi) [reply Generate.py:204]: Yeah, the creation of erargs happens after this line:

```
    if args.multi == 0:
        raise ValueError(
            "No individual player files found and number of players is 0. "
            "Provide individual player files or specify the number of players via host.yaml or --multi."
        )
```

Which to me indicates that it would be safe to just `setattr(erargs, "names", {})` right after its creation. What do you think about that?

Comment (qwint) [reply Generate.py:204]: no qualms, will do some testing around that later
